test/server/Server.test.js

```diff
@@ -1,107 +0,0 @@
-"use strict";
-
-const { relative, sep } = require("path");
-const webpack = require("webpack");
-const Server = require("../../lib/Server");
-const config = require("../fixtures/simple-config/webpack.config");
-const port = require("../ports-map").server;
-const isWebpack5 = require("../helpers/isWebpack5");
-
-const baseDevConfig = {
-  port,
-  host: "localhost",
-  static: false,
-};
-
-describe("Server", () => {
-  describe("DevServerPlugin", () => {
-    let entries;
-
-    function getEntries(server) {
-      if (isWebpack5) {
-        server.middleware.context.compiler.hooks.afterEmit.tap(
-          "webpack-dev-server",
-          (compilation) => {
-            const mainDeps = compilation.entries.get("main").dependencies;
-            const globalDeps = compilation.globalEntry.dependencies;
-
-            entries = globalDeps
-              .concat(mainDeps)
-              .map((dep) => relative(".", dep.request).split(sep));
-          }
-        );
-      } else {
-        entries = server.middleware.context.compiler.options.entry.map((p) =>
-          relative(".", p).split(sep)
-        );
-      }
-    }
-
-    it("add hot option", (done) => {
-      const compiler = webpack(config);
-      const server = new Server({ ...baseDevConfig, hot: true }, compiler);
-
-      compiler.hooks.done.tap("webpack-dev-server", () => {
-        expect(entries).toMatchSnapshot();
-
-        server.stopCallback(done);
-      });
-
-      server.startCallback((error) => {
-        if (error) {
-          throw error;
-        }
-
-        getEntries(server);
-      });
-    });
-
-    it("add hot-only option", (done) => {
```

Comment:
e2e tests already present for `hot: true` and `hot: 'only'` - https://github.com/webpack/webpack-dev-server/blob/master/test/e2e/hot-and-live-reload.test.js